src/lib/storage/chunk.cpp

```diff
@@ -254,7 +254,8 @@ void Chunk::set_individually_sorted_by(const std::vector<SortColumnDefinition>&
 }
 
 std::optional<CommitID> Chunk::get_cleanup_commit_id() const {
-  if (_cleanup_commit_id == CommitID{0}) {
+  // TODO(Martin): check with PR #2402 whether GCC still needs the `.load()` (some used in abstract_task.cpp twice).
```

Comment:
Highlighting it so that this open point is not forgotten in this PR's large diff.

Reply:
I added a todo in the PR description of #2402.